pkg/proxy/conntrack/conntrack.go

```diff
@@ -60,6 +60,26 @@ func ClearEntriesForIP(execer exec.Interface, ip string, protocol v1.Protocol) e
 	return nil
 }
 
+// ClearEntriesForIPNoNAT uses the conntrack tool to delete the conntrack entries
+// for the UDP connections that use the given service IP as destination IP and
+// don't get DNATed. It's used in the following scenario:
+// When a packet arrives, the rules for the service IP may not be installed yet,
+// the packet will be forwarded without NAT, producing a conntrack entry which
+// keeps black hole subsequent packets of the connection even after the rules for
+// the service IP are installed. The conntrack entry must be deleted to make it go
+// thought NAT again so that an endpoint can be selected.
+func ClearEntriesForIPNoNAT(execer exec.Interface, ip string, protocol v1.Protocol) error {
+	parameters := parametersWithFamily(utilnet.IsIPv6String(ip), "-D", "--orig-dst", ip, "--reply-src", ip, "-p", protoStr(protocol))
```

Comment:
`--reply-src` / `--reply-port-src` seem like a weird way to distinguish black hole from non-black hole...

I guess the issue is that we really want to say "match connections that _don't_ have a DNAT mapping", but you can't say that, you can only say "match connections that _do_ have a DNAT mapping" (`--dst-nat`)? I guess that's another thing that using netlink could fix...

Reply:
Right, this is not reliable. Will check if there are other ways to do it, like netlink you suggested.